spec/GraphQLOverHTTP.md

```diff
@@ -112,72 +178,152 @@ http://example.com/product/graphql
 
-A server MUST support requests from clients with HTTP header `Content-Type: application/graphql+json` indicating that the body of the request is a JSON document with a GraphQL request. A server must indicate the content type of the response with a `Content-Type` header. This default value should be `Content-Type: application/graphql+json` indicating that the response is a valid JSON document [conforming to the GraphQL spec](https://spec.graphql.org/October2021/#sec-Response-Format).
+Note: These defaults are in place to maintain compatibility with legacy clients.
 
-A server MAY support requests from clients with other content types.
+Note: Prior to this specification, the media type `application/json` was in wide
+use for both the HTTP request body type and the HTTP response payload type.
+Unfortunately this causes a number of issues, not least that it means clients
+cannot trust responses from the server that do not use an HTTP 2xx status code
+(since these replies may come from non-compliant HTTP servers or proxies
+somewhere in the request chain). For this reason, this specification introduces
+the `application/graphql+json` media type; however, to give existing servers
+time to move over, it is not required to be supported until 1st January 2025.
 
-A client MUST handle receiving responses with HTTP header `Content-Type: application/graphql+json` since any compliant server must support this type.
+### Legacy watershed
 
-This header MAY include encoding information (e.g. `Content-Type: application/graphql+json; charset=utf-8`)
+From 1st January 2025 (`2025-01-01T00:00:00Z`), a server MUST support requests
+encoded with the `application/graphql+json` media type (as indicated by the
+`Content-Type` header).
+
+From 1st January 2025 (`2025-01-01T00:00:00Z`), a server MUST support requests
+which accept the `application/graphql+json` media type (as indicated by the
+`Accept` header).
 
 # Request
 
-A server MUST accept POST requests, and MAY accept other HTTP methods, such as GET.
+A server MUST accept POST requests, and MAY accept other HTTP methods, such as
+GET.
 
 ## Request Parameters
 
-A request for execution should contain the following request parameters:
+A {GraphQL-over-HTTP request} is formed of the following parameters:
+
+- {query} - A Document containing GraphQL Operations and Fragments to execute.
+- {operationName} - (_Optional_): The name of the Operation in the Document to
+  execute.
+- {variables} - (_Optional_): Values for any Variables defined by the Operation.
+- {extensions} - (_Optional_): This entry is reserved for implementors to extend
+  the protocol however they see fit.
+
+Note: When comparing against
+[request in the GraphQL specification](https://spec.graphql.org/draft/#request)
+you should note the schema and "initial value" are not included in the
+GraphQL-over-HTTP request, they are handled by the server based on the URL used.
+
+Note: Be aware that `query` is a misleading parameter name as its value is a
```

Comment:
Should we not introduce a new term in this case `document` and support `query` as deprecated. Also, this discussion should go in a separate PR. But something we should think about.

Reply:
I was seriously tempted to make it so that `application/graphql+json` uses the term "document" (or maybe "source") whereas `application/json` uses "query", but decided that this would probably be ill advised. So yeah, definitely a follow-up discussion for that 👍